openrewrite/src/javascript/parser.ts

```diff
@@ -549,7 +550,19 @@ export class JavaScriptParserVisitor {
     }
 
     visitTypeParameter(node: ts.TypeParameterDeclaration) {
-        return this.visitUnknown(node);
+        if (node.constraint || (node.modifiers && node.modifiers.length) || node.default) {
+            return this.visitUnknown(node);
+        }
+
+        return new J.TypeParameter(
+            randomId(),
+            this.prefix(node),
+            Markers.EMPTY,
+            [],
+            [],
+            this.visit(node.name),
+            null
```

Comment:
Nice. Adding the constraints can be done separately. Unlike Java I think TS allows specifying anonymous types inline as in:

```ts
type Animal = { name: string };
type Dog = { breed: string } & Animal;

function handleAnimal<T extends Animal>(animal: T) {
    console.log(animal.name);
}

function handleAnimals<T extends { breed: string } & Animal>(animal: T) {
    console.log(animal.name);
}
```

Reply:
yup. and for now idk how to solve that properly

Reply:
I think the parsing will not be an issue, ad the property is of type TypeTree. We just need to implement the mapping.

Reply:
right, ok